clients/src/main/java/org/apache/kafka/common/config/ConfigDef.java

```diff
@@ -961,6 +960,75 @@ public String toRst() {
     }
 
     /**
+     * Get the configs with new metadata(group, orderInGroup, dependents) formatted with reStructuredText, suitable for embedding in Sphinx
+     * documentation.
+     */
+    public String toEnrichedRst() {
+        List<ConfigKey> configs = sortedConfigsInGroup();
+        StringBuilder b = new StringBuilder();
+        String lastKeyGroupName = "";
+
+        for (ConfigKey def : configs) {
+
+            if (!def.group.equalsIgnoreCase(lastKeyGroupName)) {
+                b.append(def.group);
+                b.append("\n");
+                char[] underLine = new char[def.group.length()];
+                Arrays.fill(underLine, '^');
+                b.append(new String(underLine));
+                b.append("\n\n");
+            }
+            lastKeyGroupName = def.group;
+
+            b.append("``");
+            b.append(def.name);
+            b.append("``\n");
+
+            for (String docLine : def.documentation.split("\n")) {
```

Comment:
At this point we have a lot of repeated code with `toRst()`. Would probably make sense to refactor to share the majority of the code for each config's main set of fields.